projects/timer_hw/src/main.c

```diff
@@ -0,0 +1,51 @@
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+#define COUNTER_PERIOD_MS 500
+
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+
+void prv_timer_callback_2(SoftTimerId timer_id, void *context);
```

Comment:
Don't worry about forward declaring your functions, just declare them where you write the function bodies.
This line can go